server/routes/settings/notifications.ts

```diff
@@ -377,4 +378,88 @@ notificationRoutes.post('/lunasea/test', async (req, res, next) => {
   }
 });
 
+notificationRoutes.get('/gotify', (_req, res) => {
+  const settings = getSettings();
+
+  res.status(200).json(settings.notifications.agents.gotify);
+});
+
+notificationRoutes.post('/gotify', (req, rest) => {
+  const settings = getSettings();
+
+  settings.notifications.agents.gotify = req.body;
+  settings.save();
+
+  rest.status(200).json(settings.notifications.agents.gotify);
+});
+
+notificationRoutes.post('/gotify/test', async (req, rest, next) => {
+  if (!req.user) {
+    return next({
+      status: 500,
+      message: 'User information is missing from request',
+    });
+  }
+
+  const gotifyAgent = new GotifyAgent(req.body);
+  if (
+    await gotifyAgent.send(Notification.TEST_NOTIFICATION, {
+      notifyAdmin: false,
+      notifyUser: req.user,
+      subject: 'Test Notification',
+      message:
+        'This is a test notification! Check check, 1, 2, 3. Are we coming in clear?',
+    })
+  ) {
+    return rest.status(204).send();
+  } else {
+    return next({
+      status: 500,
+      message: 'Failed to send gotify notification.',
+    });
+  }
+});
+
+notificationRoutes.get('/gotify', (_req, res) => {
+  const settings = getSettings();
+
+  res.status(200).json(settings.notifications.agents.gotify);
+});
+
+notificationRoutes.post('/gotify', (req, rest) => {
+  const settings = getSettings();
+
+  settings.notifications.agents.gotify = req.body;
+  settings.save();
+
+  rest.status(200).json(settings.notifications.agents.gotify);
+});
+
+notificationRoutes.post('/gotify/test', async (req, rest, next) => {
+  if (!req.user) {
+    return next({
+      status: 500,
+      message: 'User information is missing from request',
+    });
+  }
+
+  const gotifyAgent = new GotifyAgent(req.body);
+  if (
+    await gotifyAgent.send(Notification.TEST_NOTIFICATION, {
+      notifyUser: req.user,
+      notifyAdmin: false,
+      subject: 'Test Notification',
+      message:
+        'This is a test notification! Check check, 1, 2, 3. Are we coming in clear?',
+    })
+  ) {
+    return rest.status(204).send();
+  } else {
+    return next({
+      status: 500,
+      message: 'Failed to send gotify notification.',
+    });
+  }
+});
+
```

Comment:
Why are these duplicated?  Also, be sure to capitalize "Gotify" in the test failure message.

Reply:
Corrected in 59ac8